### ISSUE TITLE
Feature request: Add output_padding for deconv4

### ISSUE THREAD

**Issue by ngphuoc** [denizyuret/Knet.jl#477]
I intended to use this feature when translating code from pytorch and realized that Knet hasn't got it. 

"The padding argument effectively adds dilation * (kernel_size - 1) - padding amount of zero padding to both sizes of the input. This is set so that when a Conv2d and a ConvTranspose2d are initialized with same parameters, they are inverses of each other in regard to the input and output shapes. However, when stride > 1, Conv2d maps multiple input shapes to the same output shape. output_padding is provided to resolve this ambiguity by effectively increasing the calculated output shape on one side. Note that output_padding is only used to find output shape, but does not actually add zero-padding to output."

Ref: https://pytorch.org/docs/stable/nn.html?highlight=conv#torch.nn.ConvTranspose2d

**Comment by denizyuret** [comment]
I do not understand how deconv4 and dilation is supposed to interact, so bringing @ilkerkesen in.
